documentation/modules/advanced/pages/rbac.adoc

```diff
@@ -0,0 +1,211 @@
+----
+
+Let's send a request now:
+
+[source, bash]
+----
+curl customer-tutorial.$(minishift ip).nip.io
+
+customer => preference => recommendation v1 from 'b4d67bcb7-7rp88': 6
+----
+
+The communication now is possible.
+
+Now open link:{github-repo}/{istiofiles-dir}/namespace-rbac-policy.yml[istiofiles/namespace-rbac-policy.yml] file and change the `method` field form `GET` to `POST` and save it.
+
+[source, yaml]
+.istiofiles/namespace-rbac-policy.yml
+----
+spec:
+  rules:
+  - services: ["*"]
+    methods: ["POST"]
+----
+
+And update the Istio resource in the mesh:
+
+[source, bash,subs="+macros,+attributes"]
+----
+istioctl replace -f link:{github-repo}/{istiofiles-dir}/namespace-rbac-policy.yml[istiofiles/namespace-rbac-policy.yml] -n tutorial
+----
+
+And if you run again:
+
+[source, bash]
+----
+curl customer-tutorial.$(minishift ip).nip.io
+
+RBAC: access denied
+----
+
+Obviously, you'll get an access denied error since all requests in our application is done using `GET` method.
+
+Restore again the `method` value to `GET` and update the Istio resouce in the mesh:
+
+[source, bash,subs="+macros,+attributes"]
+----
+istioctl replace -f link:{github-repo}/{istiofiles-dir}/namespace-rbac-policy.yml[istiofiles/namespace-rbac-policy.yml] -n tutorial
+----
+
+Validate that now the communication is granted.
+
+[source, bash]
+----
+curl customer-tutorial.$(minishift ip).nip.io
+
+customer => preference => recommendation v1 from 'b4d67bcb7-7rp88': 6
+----
+
+[#authorization-jwt]
+=== Authorization and JWT
+
+In <<Granting Access>>, you have added all users access to all services, but you can also specify a list of `subjects` that are assigned to a role.
+Istio supports several ways to specify them, from using a Kubernetes service account to a range of IPs, request headers or request auths.
+But one that it is really interesting is its integration with JWT.
+
+In this section, you'll learn how to use a JWT claim to manage the access to the services.
+
+The first thing you need to do is applying all the steps described at xref:jwt.adoc[End-user authentication with JWT] section so you enable JWT authentication logic inside the mesh.
+
+After that, you are ready to use JWT in the authorization phase as well.
+
+Now apply next Istio resource which makes that only tokens that contain a field named `preferred_username` with value `demo` (notice that when you were setting up Keycloak you created this user), then the access is granted, in any other cases, the access is rejected.
+
+[source, bash,subs="+macros,+attributes"]
+----
+istioctl replace -f link:{github-repo}/{istiofiles-dir}/namespace-rbac-policy-jwt.yml[istiofiles/namespace-rbac-policy-jwt.yml] -n tutorial
+----
+
+and then you can do a new call with the token:
+
+[source, bash]
+----
+token="eyJhbGciOiJSUzI1NiIsInR5cCIg....."
+
+curl -H "Authorization: Bearer $token" customer-tutorial.$(minishift ip).nip.io
```

Comment:
I don't know why I'm getting Access Denied. I'm using the following token:

eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJRdlBDeERjYXozWm9qVzgyMm1LMDJ6dUVEN0I5aFQ0TlRPU09OU01UT1VrIn0.eyJqdGkiOiJhOWJjMGZkNi00MmZjLTQyY2YtOTM1Yy0xYTQ3NTM0ZDQ5OTgiLCJleHAiOjE1NDE3NzU5MjEsIm5iZiI6MCwiaWF0IjoxNTQxNzc1NjIxLCJpc3MiOiJodHRwOi8va2V5Y2xvYWstdHV0b3JpYWwuMTkyLjE2OC42NC4xMi5uaXAuaW8vYXV0aC9yZWFsbXMvaXN0aW8iLCJhdWQiOiJjdXN0b21lci10dXRvcmlhbCIsInN1YiI6ImYwZWIyYzQ5LTEyMTYtNDUzOC1hNzU0LTUxMDI4YzcyMmU1YyIsInR5cCI6IkJlYXJlciIsImF6cCI6ImN1c3RvbWVyLXR1dG9yaWFsIiwiYXV0aF90aW1lIjowLCJzZXNzaW9uX3N0YXRlIjoiNDFkOTZlMjUtNTA0MC00ZjhmLWE1N2QtNTgyZjA2NTA2ODFiIiwiYWNyIjoiMSIsImFsbG93ZWQtb3JpZ2lucyI6W10sInJlYWxtX2FjY2VzcyI6eyJyb2xlcyI6WyJvZmZsaW5lX2FjY2VzcyIsInVtYV9hdXRob3JpemF0aW9uIiwidXNlciJdfSwicmVzb3VyY2VfYWNjZXNzIjp7ImFjY291bnQiOnsicm9sZXMiOlsibWFuYWdlLWFjY291bnQiLCJtYW5hZ2UtYWNjb3VudC1saW5rcyIsInZpZXctcHJvZmlsZSJdfX0sInNjb3BlIjoicHJvZmlsZSBlbWFpbCIsImVtYWlsX3ZlcmlmaWVkIjpmYWxzZSwicHJlZmVycmVkX3VzZXJuYW1lIjoiZGVtbyJ9.Myyp73Wr8EepbPfo8rXtuBlvgeZDGAR3rLXuVUMiSg2cGAhEjG_G6Y5TkgEReC6fAYV2Al6OWgi-qBg2pO4TqDGL8J0HnuG1yyVGOy2Ruf261_GjkSW0QwcIIW4DFQ6QSUV22efslca_DGD7czl8C67kTGFf6Aky5uUFPb4izmE1MqKcMfYJe2ZeJoSpaEwmSHP4DkOO9FHS7B-FOBanMJo97nDN7T7Samu8l0IC8L348h8GDWdEk8pQfdQScxRnJgdqBLUh36EwqJod4i3Xw7s-_FJUyasXZe7o_bKc-HxwwMKxrj8WUzrmjUArnnAqSnbjrup8o2aKFVylVXZwEw

Reply:
Well you need to generate one token after the installation, also have you applied all the JWT section? You need to enable JWT and apply Istio resources, not just install Keycloak

Reply:
I think I've solved. Hold on, please.